src/classes/UTIL_Describe.cls

```diff
@@ -865,4 +866,33 @@ public class UTIL_Describe {
 
         return recTypeId;
     }
+    /*******************************************************************************************************
+    * @description Cache for SObject type to List of ChildRelationshipNames
+    */
+    private static Map<SObjectType, List<String>> sObjectTypeToChildRelationshipNames = new Map<SObjectType, List<String>>();
+    
+
+    /*******************************************************************************************************
+    * @description Returns the list of all Child Relationship Name for a given Sobject Type
+    * @return sObjectTypeToChildRelationshipNames.get(sObjectTypeName) List of all the child Relationship Names
+    */
+    public static List<String> getAllChildRelationshipNames(SobjectType sObjectTypeName) {
+        if (sObjectTypeToChildRelationshipNames.get(sObjectTypeName) == NULL) {
```

Comment:
@skosgi1 Please use `UTIL_Describe.` before `sObjectTypeToChildRelationshipNames` to be explicit that we are interacting with a static variable.